### PR TITLE
Zvfhmin, Zvfbfmin, Zvfbfwma 100% Coverage

### PR DESCRIPTION
**Pull request by fourth-bit**
Added templates for vector bf16 covergroups, modified test plans, and some small changes to edge values for the bf16 cases. Coverage is at 100%

### CODE REVIEW

File: framework/src/act/fcov/disassemble.svh [hunk 1]
```diff
@@ -1422,6 +1422,15 @@ function string disassemble (logic [31:0] instrRaw);
     VSHA2CH_VV:   $sformat(decoded, "vsha2ch.vv %s, %s, %s",              vd, vs2, vs1);
     VSHA2CL_VV:   $sformat(decoded, "vsha2cl.vv %s, %s, %s",              vd, vs2, vs1);
 
+    // V Extension: BF16 Minimal Instructions
```
Comment (jordancarlin) [inline framework/src/act/fcov/disassemble.svh]: Update the comment to indicate the actual extension name. This is not part of the V extension.

File: coverpoints/unpriv/Zvfbfwma_coverage_init.svh [hunk 2]
```diff
@@ -10,5 +10,9 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 ////////////////////////////////////////////////////////////////////////////////////////////////
+`ifndef ELEN16
     Zvfbfwma_vfwmaccbf16_vf_cg = new(); Zvfbfwma_vfwmaccbf16_vf_cg.set_inst_name("obj_Zvfbfwma_vfwmaccbf16_vf");
+`endif
```
Comment (jordancarlin) [inline coverpoints/unpriv/Zvfbfwma_coverage_init.svh]: Why is this all surrounded in `ifndef ELEN16?  Aren't instructions that operate on 16 bit elements?

**Review by jordancarlin — COMMENTED** [review]
A few minor comments.

File: coverpoints/unpriv/Zvfbfwma_coverage_init.svh (hunk 2 above)
Comment (fourth-bit) [reply coverpoints/unpriv/Zvfbfwma_coverage_init.svh]: That's odd. It looks like this is what is already happening in the Vf16 for `vfwmacc`. Should the template in `init_vector_widen.sv` have `ifndef ELENTWOEFFEW` instead?

Comment (jordancarlin) [reply coverpoints/unpriv/Zvfbfwma_coverage_init.svh]: @jacassidy would know best.

File: generators/testgen/scripts/vector-testgen-unpriv.py [hunk 3]
```diff
@@ -909,6 +932,7 @@ def makeTest(coverpoints, test, sew=None):
     elif coverpoint == "cp_rs2_edges_ls_e64"        : make_rs2_edges_v(test, sew, redges_ls_e64, lmul = getBaseLmul(test, sew))
     elif coverpoint == "cp_rs1_edges"               : make_rs1_edges_v(test, sew, redgesv)
     elif coverpoint == "cp_fs1_edges_v"             : make_fs1_edges_v(test, sew)
+    elif coverpoint == "cp_fs1_edges_v_bf16"        : make_fs1_edges_v(test, sew)
```
[the remark this answers is not in the document]
Comment (fourth-bit) [reply generators/testgen/scripts/vector-testgen-unpriv.py:947]: Handled by making a list of bf16 instructions instead

File: generators/testgen/scripts/vector-testgen-unpriv.py [hunk 4]
```diff
@@ -1104,7 +1132,7 @@ def makeTest(coverpoints, test, sew=None):
     elif coverpoint == "cp_custom_vshiftn_upperbits_rs1_ones"         : make_custom_vshift_upperbits_r1_ones(test, sew, "rs1", narrow=True)
     elif coverpoint == "cp_custom_vindexedges_index_ge_vlmax"         : make_custom_vindexedges_index_ge_vlmax(test, sew)
     elif coverpoint == "cp_custom_vindexedges_index_gt_vl_lt_vlmax"   : make_custom_vindexedges_index_gt_vl_lt_vlmax(test, sew)
-    elif coverpoint[:2] != "cp"                                       : pass # skip all the helper coverpoints
+    elif coverpoint[:2] not in ("cp", "cr")                           : pass # skip all the helper coverpoints
     elif coverpoint in REGISTRY                                       : setCurrentCoverpoint(coverpoint); REGISTRY[coverpoint](test, sew)   # call the registered function (cp_custom_**)
```
[the remark this answers is not in the document]
Comment (fourth-bit) [reply generators/testgen/scripts/vector-testgen-unpriv.py:1148]: Not implementing a cross should be an error, and this currently emits no warnings on any of Vx, Vls, and Vf.

File: coverpoints/unpriv/Zvfbfwma_coverage.svh [hunk 5]
```diff
@@ -19,19 +19,18 @@
...
     cp_asm_count : coverpoint ins.ins_str == "vfwmaccbf16.vf"  iff (ins.trap == 0 )  {
         // Number of times instruction is executed
         bins count[]  = {1};
     }
 
-    cp_csr_fflags_vun : coverpoint get_csr_val(ins.hart, ins.issue, `SAMPLE_AFTER, "fcsr", "fflags") iff (ins.trap == 0 )  {
+    cp_csr_fflags_vn : coverpoint get_csr_val(ins.hart, ins.issue, `SAMPLE_AFTER, "fcsr", "fflags") iff (ins.trap == 0 )  {
         // Value of FCSR.fflags
         wildcard bins NV   = (5'b0???? => 5'b1????);
         wildcard bins NV1  = (5'b1???? => 5'b1????);
```
[the remark this answers is not in the document]
Comment (fourth-bit) [reply coverpoints/unpriv/Zvfbfwma_coverage.svh:32]: Changed to voun

File: coverpoints/unpriv/Zvfbfwma_coverage_init.svh (hunk 2 above)
Comment (jacassidy) [reply coverpoints/unpriv/Zvfbfwma_coverage_init.svh]: That is very strange I will be able to take a look in a bit but that might have bren a mistake from priv, ryans suggestion is correct ill figure out why it generated incorrectly

Comment (jacassidy) [reply coverpoints/unpriv/Zvfbfwma_coverage_init.svh]: Took a look and it seems like theres a template for vector widening with those strange ifdefs, doesnt seem like they do anything because widening is initialized with eew = 2*sew so ill pr removing them, no need to hold up this pr @jordancarlin

File: coverpoints/unpriv/Zvfhmin_coverage.svh [hunk 6]
```diff
@@ -178,9 +350,16 @@ function void zvfhmin_sample(int hart, int issue, ins_t ins);
     if (get_csr_val(hart, issue, `SAMPLE_BEFORE, "vtype", "vsew") == 1 ||
         get_csr_val(hart, issue, `SAMPLE_BEFORE, "vtype", "vill") == 1) begin
         case (traceDataQ[hart][issue][0].inst_name)
-        "vfwcvt.f.f.v"     : begin
-            Zvfhmin_vfwcvt_f_f_v_cg.sample(ins);
-        end
+        `ifndef ELEN16
```
Comment (davidharrishmc) [inline coverpoints/unpriv/Zvfhmin_coverage.svh]: Why are these ifndef instead of ifdef, and why don't we know that ELEN16 is always defined if Zvfhmin is supported?

**Review by davidharrishmc — APPROVED** [review]
